test/staking/SimpleDAIStaking.test.ts

```diff
@@ -792,354 +772,352 @@ contract("SimpleDAIStaking - staking with DAI mocks", ([founder, staker]) => {
+      cDAI.address
+    ]);
+    const ictrl = await ethers.getContractAt(
+      "Controller",
+      controller,
+      schemeMock
+    );
+    await ictrl.genericCall(
+      goodCompoundStaking.address,
+      encodedCall,
+      avatar,
+      0
+    );
+    let avatarBalanceAfter = await cDAI.balanceOf(avatar);
+    let stakingBalance = await cDAI.balanceOf(goodCompoundStaking.address);
     // checks that something was recovered
     expect(avatarBalanceAfter.sub(avatarBalanceBefore).toString()).to.not.equal("0");
     expect(stakingBalance.toString()).to.be.equal("0");
   });
 
   it("should not transfer any funds if trying to execute recover of a token without balance", async () => {
-    const cdai1 = await cDAIMock.new(dai.address);
-    let simpleStaking1 = await GoodCompoundStaking.new(
+    const cdaiFactory = await ethers.getContractFactory("cDAIMock")
+    const cdai1 = await cdaiFactory.deploy(dai.address);
+    const goodCompoundStakingFactory = await ethers.getContractFactory(
+      "GoodCompoundStaking"
+    );
+
+    let simpleStaking1 = await goodCompoundStakingFactory.deploy(
       dai.address,
       cdai1.address,
-      founder,
       BLOCK_INTERVAL,
-      avatar.address,
-      identity.address
+      controller,
+      nameService.address
     );
-    await dai.mint(founder, web3.utils.toWei("100", "ether"));
-    await dai.approve(cdai1.address, web3.utils.toWei("100", "ether"));
-    await cdai1.balanceOf(founder);
-    await cdai1.mint(web3.utils.toWei("100", "ether"));
+    await dai["mint(address,uint256)"](founder.address, ethers.utils.parseEther("100"));
+    await dai.approve(cdai1.address, ethers.utils.parseEther("100"));
+    await cdai1.balanceOf(founder.address);
+    await cdai1["mint(uint256)"](ethers.utils.parseEther("100"));
     await cdai1.transfer(simpleStaking1.address, "0");
-    let balanceBefore = await cdai1.balanceOf(avatar.address);
-    let encodedCall = web3.eth.abi.encodeFunctionCall(
-      {
-        name: "recover",
-        type: "function",
-        inputs: [
-          {
-            type: "address",
-            name: "_token"
-          }
-        ]
-      },
-      [cdai1.address]
+    let balanceBefore = await cdai1.balanceOf(avatar);
+    
+    let encodedCall = goodCompoundStakingFactory.interface.encodeFunctionData("recover", [
+      cdai1.address
+    ]);
+    const ictrl = await ethers.getContractAt(
+      "Controller",
+      controller,
+      schemeMock
     );
-    await controller.genericCall(simpleStaking1.address, encodedCall, avatar.address, 0);
-    let balanceAfter = await cdai1.balanceOf(avatar.address);
+    await ictrl.genericCall(
+      simpleStaking1.address,
+      encodedCall,
+      avatar,
+      0
+    );
+    let balanceAfter = await cdai1.balanceOf(avatar);
     expect(balanceAfter.toString()).to.be.equal(balanceBefore.toString());
   });
 
-  it("should not be able to change the fund manager address if not owner", async () => {
```

Comment:
can we test that if fundmanager changes in nameservice it still works as expected?